frontend/deploy.sh

```diff
@@ -0,0 +1,250 @@
+            exit 1
+        fi
+    done
+}
+
+function populateRequiredParams () {
+    local paramsFile=$1
+    printf "Checking required parameters... "
+    checkRequiredParams $paramsFile
+    # The jq command below sets environment variables based on the key-value pairs in a JSON-formatted file
+    eval $(jq -r 'to_entries | .[] | "export \(.key)=\(.value)"' $paramsFile)
+    printf "Done.\n"
+}
+
+function set_variables() {
+    SUBSCRIPTION_ID=${SUBSCRIPTION_ID:-""}
+    RESOURCE_GROUP=${RESOURCE_GROUP:-""}
+    LOCATION=${LOCATION:-""}
+    AAD_CLIENT_ID=${AAD_CLIENT_ID:-""}
+    AAD_OBJECT_ID=${AAD_OBJECT_ID:-""}
+    AAD_TENANT_ID=${AAD_TENANT_ID:-""}
+    AAD_TOKEN_ISSUER_URL=${AAD_TOKEN_ISSUER_URL:-"https://login.microsoftonline.com/$AAD_TENANT_ID/v2.0"}
+    IMAGE_NAME=${IMAGE_NAME:-"graphrag:frontend"}
+    REGISTRY_NAME=${REGISTRY_NAME:-"${RESOURCE_GROUP}reg"}
+    APP_SERVICE_PLAN=${APP_SERVICE_PLAN:-"${RESOURCE_GROUP}-asp"}
+    WEB_APP=${WEB_APP:-"${RESOURCE_GROUP}-playground"}
+    WEB_APP_IDENTITY=${WEB_APP_IDENTITY:-"${WEB_APP}-identity"}
+    #BACKEND_RESOURCE_GROUP=${BACKEND_RESOURCE_GROUP:-""} # needed for backend outbound vnet integration
+}
+function create_resource_group {
+    az account set --subscription $SUBSCRIPTION_ID
+    az group create --name $RESOURCE_GROUP --location $LOCATION
+}
+
+function create_acr() {
+    az acr create --resource-group $RESOURCE_GROUP \
+    --name $REGISTRY_NAME \
+    --sku Basic \
+    --admin-enabled true
+}
+
+function login_to_acr() {
+    az acr login --name $REGISTRY_NAME
+}
+
+function build_and_push_image() {
+    local SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+    az acr build --registry $REGISTRY_NAME -f $SCRIPT_DIR/../docker/Dockerfile-frontend --image $IMAGE_NAME $SCRIPT_DIR/../
+}
+
+function create_app_service_plan() {
+    az appservice plan create --name $APP_SERVICE_PLAN \
+        --resource-group $RESOURCE_GROUP \
+        --sku B3 \
+        --is-linux
+}
+
+function create_web_app() {
+    az webapp create --resource-group $RESOURCE_GROUP \
+        --plan $APP_SERVICE_PLAN \
+        --name $WEB_APP \
+        --deployment-container-image-name $REGISTRY_NAME.azurecr.io/$IMAGE_NAME
+}
+
+function create_web_app_identity() {
+    IDENTITY_RESULT=$(az identity create --resource-group $RESOURCE_GROUP --name $WEB_APP_IDENTITY --output json)
+    WEBAPP_IDENTITY_ID=$(jq -r .id <<< $IDENTITY_RESULT)
+    WEBAPP_IDENTITY_OBJECT_ID=$(jq -r .principalId <<< $IDENTITY_RESULT)
+    WEBAPP_IDENTITY_CLIENT_ID=$(jq -r .clientId <<< $IDENTITY_RESULT)
+    az webapp identity assign --name $WEB_APP \
+        --resource-group $RESOURCE_GROUP \
+        --identities $WEBAPP_IDENTITY_ID
+}
+
+function configure_registry_credentials() {
+    ACR_ID=$(az acr show --name $REGISTRY_NAME --resource-group $RESOURCE_GROUP --query id --output tsv)
+    az role assignment create --assignee $WEBAPP_IDENTITY_CLIENT_ID \
```

Comment:
I get the following error at this step in the script

```
Cannot find user or service principal in graph database for '<webapp_identity_client_idenity>`. If the assignee is an appId, make sure the corresponding service principal is created with 'az ad sp create --id '<webapp_identity_client_idenity>'.'
```
Except the webapp client identity is filled out and correct. I verified in the portal.

Reply:
Issue identified as `az-cli` configuration not outputting json as needed for WEBAPP_IDENTITY_CLIENT_ID. Script updated to force json output. Issue resolved.